tex/gregoriotex-signs.tex

```diff
@@ -1644,19 +1644,21 @@
 % #3 is the char of the alteration hole
 % #4 is 1 in the case of a flat for a key change, 0 otherwise
 \def\gre@alteration#1#2#3#4{%
-  % to see why we reset \gre@lastoflinecount, see comments of \GreGlyph.
+  \setbox\gre@box@temp@width=\hbox{#2}%
   \ifnum#4=0\relax %
     \gre@newglyphcommon %
+    \ifgre@firstglyph%
+      \global\gre@dimen@bolextra = \wd\gre@box@temp@width%
+      \gre@skip@temp@four = \gre@skip@alterationspace\relax%
+      \global\advance\gre@dimen@bolextra by \dimexpr\gre@skip@temp@four\relax%
+      \gre@debugmsg{bolshift}{bolextra: \the\ger@diment@bolextra}%
+%			\kern -\gre@dimen@bolextra\relax%
+    \fi%
```

Comment:
`bolextra` depends on the size of the alteration and thus needs to be computed in a location where GregorioTeX knows what sort alteration is being printed.  This information is only available inside the macro which prints the alteration, and is not currently available to `\gre@calculate@bolshift`.  I thus have to pass that information out from `\gre@alteration`, which I do via `bolextra`, which is then used in the calculation of `bolshift`.

You might be confused by the `\kern` line in the portion of the code you're looking at.  That's a left over from my testing of where the best place to incorporate the extra shift was and is commented out.  I'll delete it in order to save the confusion.